### ISSUE TITLE
Docker Container Run Fails

### ISSUE THREAD

**Issue by pasanchamikara** [aquasecurity/cloudsploit#1118]
While executing the docker container, the following error were observed.

```
‹master*› » docker run -it -e AWS_SECRET_ACCESS_KEY_ID=xxxxxxxxxxxx -e AWS_SECRET_ACCESS_KEY=yyyyyyyyyyyyyyyy cloudsploitc    
docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "cloudsploit-scan": executable file not found in $PATH: unknown.
```

This seems to be because of improper linking of the node modules which skipped the bin to not to be set properly to `cloudsploit-scan`

**Comment by pasanchamikara** [comment]
PR created https://github.com/aquasecurity/cloudsploit/pull/1119

**Comment by asdrubalos** [comment]
hi @pasanchamikara, by any chance are you using mac?
i solved this problem in Dockerfile (using npm link, chmod, added workdir and change entrypoint):
now work without $PATH error

**Dockerfile:**
FROM node:lts-alpine3.12

ARG PACKAGENAME=cloudsploit

COPY . /var/scan/cloudsploit/

RUN cd /var/scan \
&& mkdir reports \
&& npm init --yes \
&& npm install ${PACKAGENAME}

ENV PATH "$PATH:/var/scan/node_modules/.bin"

WORKDIR /var/scan/cloudsploit

RUN npm link index.js \
&& chmod +x /var/scan/cloudsploit/index.js

WORKDIR /var/scan/

ENTRYPOINT ["./cloudsploit/index.js","--config=/var/scan/cloudsploit/config.js"]
CMD []


Now the problem is the next error node_modules/async/dist/async.js:966 https://github.com/aquasecurity/cloudsploit/issues/1158

**Comment by pasanchamikara** [comment]
Hi @asdrubalos , no I was using ubuntu in my case, believe it was the missing `npm link` to fix the linking of the bin file afterall. So no need of a separate ./cloudsploit/index.js in that case.

**Comment by s-vx** [comment]
Greetings!
I think I found why there's a problem with the docker build here

**TL;DR** 
ARG PACKAGENAME=cloudsploit**/** fixes it. 

**The long explanation**
The value for the docker entrypoint appears to have been set to "cloudsploitscan" on 29 Jul 2019 in [Pull 198](https://github.com/aquasecurity/cloudsploit/pull/198)

Then changed again to `cloudsploit-scan` in [Pull 366](https://github.com/aquasecurity/cloudsploit/pull/366) on 10 Oct 
 2020. 

As of 4 May 2022, the file `package.json` from master contains the following directive (which makes sense with the current value in master):
```
"bin": {
    "cloudsploit-scan": "./index.js"
  },
``` 

But when it is installed on a fresh copy of master and built using `docker build` without build args, only the binary `cloudsploitscan` is present on the sytem. This issue is also referred to here:  [Issue 946](https://github.com/aquasecurity/cloudsploit/issues/946)

This appears to be caused by the in-docker commands
```
ARG PACKAGENAME=cloudsploit
npm install ${PACKAGENAME}
``` 
which reach out to the NPM registry and load the outdated `cloudsploit-0.0.1-dev1` package. 

This is observable in the generated `package-lock.json` which contains the following snippet:
```
    "node_modules/cloudsploit": {
      "version": "0.0.1-dev1",
      "resolved": "https://registry.npmjs.org/cloudsploit/-/cloudsploit-0.0.1-dev1.tgz",
[...]
      },
      "bin": {
        "cloudsploitscan": "index.js"
      }
    },
```

and is consistent with the content of `package.json` in `cloudsploit-0.0.1-dev1.tgz` at the resolved url above. 

All this to say:

Changing `ARG PACKAGENAME=cloudsploit` to 
`ARG PACKAGENAME=cloudsploit/` _(note the trailing slash)_ 
will ensure `npm install` considers the newly-copied folder `./cloudsploit/` and builds it instead of the old package.

I don't have a mean to test the `docker --build-arg PACKAGENAME=@myscope/cloudsploit` use case, maybe the issue is there too.

**Comment by johndekroon** [comment]
In the master repo it does seem to be broken once again.
